lib/charms/loki_k8s/v0/loki_push_api.py

```diff
@@ -708,7 +850,12 @@ class LokiPushApiProvider(RelationManagerBase):
     """A LokiPushApiProvider class."""
 
     def __init__(
-        self, charm, relation_name: str = DEFAULT_RELATION_NAME, *, rules_dir="/loki/rules"
+        self,
+        charm,
+        relation_name: str = DEFAULT_RELATION_NAME,
+        port: int = 3100,
+        *,
```

Comment:
We for sure know that self, charm and relation_name are here to stay, but not so sure about port.
```suggestion
        *,
        port: int = 3100,
```
Although that may break some existing code, if the port is passed positionally.

Reply:
In units with multiple workload containers we need to be able to set different ports or the promtails in the various containers will collide and only one will be able to bind the port.

Reply:
(Nvd from mobile I thought the comment was about the LogProxyConsumer)